lib/private/App/AppManager.php

```diff
@@ -820,60 +828,42 @@ public function getDefaultEnabledApps(): array {
 		return $this->defaultEnabled;
 	}
 
+	/**
+	 * @inheritdoc
+	 */
 	public function getDefaultAppForUser(?IUser $user = null, bool $withFallbacks = true): string {
-		// Set fallback to always-enabled files app
-		$appId = $withFallbacks ? 'files' : '';
-		$defaultApps = explode(',', $this->config->getSystemValueString('defaultapp', ''));
-		$defaultApps = array_filter($defaultApps);
-
-		$user ??= $this->userSession->getUser();
-
-		if ($user !== null) {
-			$userDefaultApps = explode(',', $this->config->getUserValue($user->getUID(), 'core', 'defaultapp'));
-			$defaultApps = array_filter(array_merge($userDefaultApps, $defaultApps));
-			if (empty($defaultApps) && $withFallbacks) {
-				/* Fallback on user defined apporder */
-				$customOrders = json_decode($this->config->getUserValue($user->getUID(), 'core', 'apporder', '[]'), true, flags:JSON_THROW_ON_ERROR);
-				if (!empty($customOrders)) {
-					// filter only entries with app key (when added using closures or NavigationManager::add the app is not guranteed to be set)
-					$customOrders = array_filter($customOrders, fn ($entry) => isset($entry['app']));
-					// sort apps by order
-					usort($customOrders, fn ($a, $b) => $a['order'] - $b['order']);
-					// set default apps to sorted apps
-					$defaultApps = array_map(fn ($entry) => $entry['app'], $customOrders);
-				}
-			}
-		}
-
-		if (empty($defaultApps) && $withFallbacks) {
-			$defaultApps = ['dashboard','files'];
-		}
-
-		// Find the first app that is enabled for the current user
-		foreach ($defaultApps as $defaultApp) {
-			$defaultApp = \OC_App::cleanAppId(strip_tags($defaultApp));
-			if ($this->isEnabledForUser($defaultApp, $user)) {
-				$appId = $defaultApp;
-				break;
-			}
-		}
-
-		return $appId;
+		$id = $this->getNavigationManager()->getDefaultEntryIdForUser($user, $withFallbacks);
+		$entry = $this->getNavigationManager()->get($id);
+		return (string)$entry['app'];
 	}
 
+	/**
+	 * @inheritdoc
+	 */
 	public function getDefaultApps(): array {
-		return explode(',', $this->config->getSystemValueString('defaultapp', 'dashboard,files'));
+		$ids = $this->getNavigationManager()->getDefaultEntryIds();
+
+		return array_map(function (string $id) {
+			$entry = $this->getNavigationManager()->get($id);
+			return (string)$entry['app'];
+		}, $ids);
```

Comment:
```suggestion
		return array_values(array_unique(array_map(function (string $id) {
			$entry = $this->getNavigationManager()->get($id);
			return (string)$entry['app'];
		}, $ids)));
```
Should remove duplicates, no?